### PR TITLE
refactor(dynacell): replace microssim dep with cubic.metrics.MicroMS3IM

### PR DESCRIPTION
**Pull request by alxndrkalinin**
## Summary
Stacked on #429. Drops the upstream `microssim @ git+https://github.com/juglab/microssim@8bccb17d` runtime dep and reroutes `calculate_microssim` through `cubic.metrics.MicroMS3IM` (cubic 0.7.0a4 ships a native, device-agnostic implementation).

- `applications/dynacell/src/dynacell/evaluation/metrics.py`: drop the top-of-file microssim import + `_require_microssim` helper; lazy-import `MicroMS3IM` from `cubic.metrics` inside `calculate_microssim`. `_require_cubic()` is *not* called there — it forces the GPU stack (cupy/cucim) and would regress CPU-only eval. The lazy import is the gate.
- `applications/dynacell/pyproject.toml`: bump `cubic==0.7.0a2` → `cubic==0.7.0a4`, drop the microssim git pin from `optional-dependencies.eval`.
- Workspace `[tool.uv.sources]`: **temporary** git override pointing `cubic` at the 0.7.0a4 release commit (`424343f`) until PyPI publishes. Remove the override afterward.
- `uv.lock` regenerated — no remaining `juglab/microssim` / `8bccb17d` references.
- Test stub at `tests/test_evaluation_metrics.py` keeps its monkey-patch shape and gains `cubic.metrics.MicroMS3IM = object` so future tests exercising the path don't need to re-extend it.

Design doc: `applications/dynacell/src/dynacell/evaluation/CUBIC_MICROSSIM_PORT.md` (and the upstream-side plan that drove the cubic implementation).

## Test plan
- [x] `uv lock` regenerates cleanly with the source override; lockfile contains no microssim references.
- [x] `uv sync --all-packages --all-extras` succeeds.
- [x] `uv run python -c \"from cubic.metrics import MicroMS3IM; ...\"` resolves.
- [x] `pytest applications/dynacell/tests/test_evaluation_metrics.py applications/dynacell/tests/test_evaluation_io.py` → 19 passed, 1 deselected (`test_gain_and_offset_errors_are_not_scale_invariant` is a pre-existing failure on `backbone-refactor` HEAD, unrelated to this PR).
- [x] `pytest applications/dynacell/tests/test_evaluation_pipeline.py test_evaluation_pipeline_parallel_cpu.py` → 3 passed.
- [ ] Spot-check one real eval run on a small fixture before merge.

## Follow-up
- Open issue for `applications/cytoland/examples/dl-course-exercise/solution.py:136` (still imports upstream microssim; out of scope for this PR — different app, example file).
- Drop the `[tool.uv.sources]` cubic override once 0.7.0a4 is on PyPI.

### CODE REVIEW

File: applications/dynacell/src/dynacell/evaluation/metrics.py [hunk 1]
```diff
@@ -259,8 +249,8 @@ def compute_pixel_metrics(prediction, target, spacing, fsc_kwargs=None, spectral
 
 def calculate_microssim(microssim_data):
     """Calculate MicroMS3IM scores across a collection of images."""
-    _require_microssim()
-    _require_cubic()
+    from cubic.metrics import MicroMS3IM
+
     targets = np.concatenate([img["target"] for img in microssim_data], axis=0)
     predictions = np.concatenate([img["predict"] for img in microssim_data], axis=0)
```
[the remark this answers is not in the document]
Comment (alxndrkalinin) [reply applications/dynacell/src/dynacell/evaluation/metrics.py:255]: Fixed in f6560078 — `np.where(...)` already returns a numpy array because `target[t]`/`predict[t]` come from `np.asarray(...)` upstream (pipeline.py:303-304), so the `asnumpy(...)` wrap was an identity. Dropping it removes the hidden dependency on the top-level `cubic.cuda.asnumpy` symbol.

File: applications/dynacell/src/dynacell/evaluation/metrics.py [hunk 2]
```diff
@@ -3,11 +3,6 @@
 import numpy as np
 import torch
```
[the remark this answers is not in the document]
Comment (alxndrkalinin) [reply applications/dynacell/src/dynacell/evaluation/metrics.py:4]: Fixed in fda796b6 — module docstring now says MicroMS3IM, matching the metric actually computed.
